api/v1alpha1/gateway_types.go

```diff
@@ -195,40 +192,80 @@ type GatewayStatus struct {
 
 // ListenerStatus is the status associated with each listener block.
 type ListenerStatus struct {
-	// Errors is a list of reasons why a given Listener Spec is
-	// not valid. Errors will be empty if the Spec is valid.
-	Errors []ListenerError `json:"errors"`
-
-	//
-	Address string `json:"address"`
+	// Address bound on this listener.
+	Address *ListenerAddress `json:"address"`
+	// Conditions describe the current condition of this listener.
+	Conditions []ListenerCondition `json:"conditions"`
 }
 
-// ListenerErrorReason is the type for errors associated with the
-// listener.
-type ListenerErrorReason string
+// ListenerConditionType is a type of condition associated with the listener.
+type ListenerConditionType string
 
 const (
-	// ErrListenerInvalidSpec is a generic error that is a
-	// catch all for unsupported configurations that do not match a
-	// more specific error. Implementors should try to use more
-	// specific errors instead of this one to give users and
-	// automation a more information.
-	ErrListenerInvalidSpec ListenerErrorReason = "InvalidSpec"
-	// ErrListenerBadAddress indicates the Address
-	ErrListenerBadAddress ListenerErrorReason = "InvalidAddress"
+	// ConditionInvalidListener is a generic condition that is a catch all for
```

Comment:
Why not simplify the condition types and use `Reason` and `Message` to convey details? See https://github.com/kubernetes-sigs/service-apis/pull/45/files#r370279364 for more details.

Reply:
The goal here was to follow the pattern established elsewhere in Kubernetes. Most notably, this follows the pattern used for core statuses (https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2132-L2142). This same pattern is used by API Machinery, API Extensions, as well as the Apps, Autoscaling, and Batch API groups. Although I agree that it would be nice to simplify this, I think there's likely more value in following the pattern that's already been established elsewhere in the project.